src/SFA.DAS.FAA.Web/Services/FilterBuilder.cs

```diff
@@ -66,9 +67,45 @@ private static List<string> BuildQueryParameters(GetSearchResultsRequest request
             {
                 queryParameters.Add($"DisabilityConfident={request.DisabilityConfident}");
             }
+
+            switch (request.IncludeCompetitiveSalaryVacancies)
+            {
+                case true:
+                    queryParameters.Add($"IncludeCompetitiveSalaryVacancies=true");
+                    break;
+                default:
+                    queryParameters.Remove("IncludeCompetitiveSalaryVacancies=false");
+                    break;
+            }
+            
             return queryParameters;
         }
 
+        public static string ReplaceQueryStringParam(string currentPageUrl, string paramToReplace, string newValue)
+        {
+            var urlWithoutQuery = currentPageUrl.IndexOf('?') >= 0
+                ? currentPageUrl[..currentPageUrl.IndexOf('?')]
+                : currentPageUrl;
+
+            var queryString = currentPageUrl.IndexOf('?') >= 0
+                ? currentPageUrl[currentPageUrl.IndexOf('?')..]
+                : null;
+
+            var queryParamList = queryString != null
+                ? HttpUtility.ParseQueryString(queryString)
+                : HttpUtility.ParseQueryString(string.Empty);
+
+            if (queryParamList[paramToReplace] != null)
+            {
+                queryParamList[paramToReplace] = newValue;
+            }
+            else
+            {
+                queryParamList.Add(paramToReplace, newValue);
+            }
+            return $"{urlWithoutQuery}?{queryParamList}";
+        }
+
```

Comment:
To summarise, this should work:

string ReplaceQueryStringParam(string currentPageUrl, string paramToReplace, string newValue)
{
    var uriBuilder = new UriBuilder(new Uri(currentPageUrl));
    var queryString = HttpUtility.ParseQueryString(uriBuilder.Query);
    queryString[paramToReplace] = newValue;
    uriBuilder.Query = queryString.ToString();
    return uriBuilder.ToString();
}